### DISCUSSION TITLE
FileUpload component: Display status and server response after upload on each file?

### DISCUSSION THREAD

**Discussion by sparky742** [primefaces/community#487] [PrimeReact]
I am using FileUpload in my React project and it is working well.  However, I'd like to keep the files in the (or a) list after upload, and show the status + a reference value that comes back from the server after upload on each file as information for the user.

For example, after upload of the files, the list should look something like:

File1.PDF OK Ref#12345
File2.TXT Failed
File3.TXT OK Ref#52531

Specifically, how can I display a status next to the filename, AND keep it in on screen so it doesn't get cleared?
Thanks.

**Comment by chrishj59** [comment]
I am using the custom upload feature and would like to give the user updates on how far through the file processing is possibly with a progress bar. Not sure if this is possible.

**Reply by sparky742** [reply]
You can show the upload progress of a file or multiple, by using onProgress on the FileUpload, with a function like this:

```
      const onProgress = ({  event }: {  event: FileUploadProgressEvent }) => {
      setUploadStatus(prevStatus => ({
        ...prevStatus,
        [event.originalEvent.target]: `${event.progress}%`
      }));
```

I'd like to show that but also get a response from the server and show that also (which sort of sounds like what you want to do after the upload, so upload would be 50% and the processing on the server might be the other 50%).

I've been thinking about just having another card below the FileUpload that I will populate with files that complete uploading.  I'll have a list of files with some kind of status next to them.  When they disappear from FileUpload, I'll capture them in an onUpload function and add them to an array.  Something like this:

```
const onUpload = ({ files, xhr }: { files: File[], xhr: XMLHttpRequest }) => {
      files.forEach(file => {
        setUploadStatus(prevStatus => ({
          ...prevStatus,
          [file.name]: 'Uploaded'
        }));
      });
      
      //const yourFiles = []; // array of your already uloaded files
      if (fileUploadRef.current != null) {
        (fileUploadRef.current as any).setUploadedFiles(files)
        console.log('set uploaded files',files);
      }
    };
```

This is still a work in progress and I'm open to suggestions for a better way!

**Comment by mertsincan** [comment]
@gucal @nitrogenous can you assist with this question?
